app/logic/serviceLearningCourses.py

```diff
@@ -77,17 +77,47 @@ def saveCourseParticipantsToDatabase(cpPreview: Dict[str, Dict[str, Dict[str, Li
                 print(f"Unable to save course {course}. {courseInfo['errorMsg']}")
                 continue
 
-            courseObj: Course = Course.get_or_create(
-                 courseAbbreviation = course,
-                 term = termObj, 
-                 defaults = {"CourseName" : "",
-                             "sectionDesignation" : "",
-                             "courseCredit" : "1",
-                             "term" : termObj,
-                             "status" : 4,
-                             "createdBy" : g.current_user,
-                             "serviceLearningDesignatedSections" : "",
-                             "previouslyApprovedDescription" : "" })[0]
+            termCheck = Term.get(Term.id == termObj)
+
+            checkTermOrder = termCheck.termOrder
+
+            check_for_Existing_Courses = list((Course.select()
+                                        .join(Term, on =(Course.term == Term.id))
+                                        .where((Course.courseAbbreviation == course) & (Term.termOrder <= checkTermOrder))
+                                        .order_by(Course.term.desc())
+                                        .limit(1)))
+            
+            if not check_for_Existing_Courses :
+
+                courseObj: Course = Course.create(courseName = "",
+                                sectionDesignation = "",
+                                courseAbbreviation = course,
+                                courseCredit = "1",
+                                term = termObj,
+                                status = 4,
+                                createdBy = g.current_user,
+                                serviceLearningDesignatedSections = "",
+                                previouslyApprovedDescription = "")
+        
+            else:
+                previous_data = check_for_Existing_Courses[0]
```

Comment:
Variable needs to be mixed case, and maybe more precise, `previousCourse` or `previousMatchedCourse`